sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/ConfigProperties.java

```diff
@@ -154,117 +78,23 @@ public Double getDouble(String name) {
-            })
-        // If duplicate keys, prioritize later ones similar to duplicate system properties on a
-        // Java command line.
-        .collect(
-            Collectors.toMap(
-                Map.Entry::getKey, Map.Entry::getValue, (first, next) -> next, LinkedHashMap::new));
-  }
-
-  /**
-   * Returns a boolean-valued configuration property. Uses the same rules as {@link
-   * Boolean#parseBoolean(String)} for handling the values.
+   * Returns a list-valued configuration property. The format of the original value must be
+   * comma-separated. Empty values will be removed.
    *
-   * @return false if the property has not been configured.
+   * @return an empty list if the property has not been configured.
    */
-  public boolean getBoolean(String name) {
-    return Boolean.parseBoolean(config.get(name));
-  }
-
-  private static ConfigurationException newInvalidPropertyException(
-      String name, String value, String type) {
-    throw new ConfigurationException(
-        "Invalid value for property " + name + "=" + value + ". Must be a " + type + ".");
-  }
-
-  private static List<String> filterBlanksAndNulls(String[] values) {
-    return Arrays.stream(values)
-        .map(String::trim)
-        .filter(s -> !s.isEmpty())
-        .collect(Collectors.toList());
-  }
-
-  /** Returns the TimeUnit associated with a unit string. Defaults to milliseconds. */
-  private static TimeUnit getDurationUnit(String unitString) {
-    switch (unitString) {
-      case "": // Falllthrough expected
-      case "ms":
-        return TimeUnit.MILLISECONDS;
-      case "s":
-        return TimeUnit.SECONDS;
-      case "m":
-        return TimeUnit.MINUTES;
-      case "h":
-        return TimeUnit.HOURS;
-      case "d":
-        return TimeUnit.DAYS;
-      default:
-        throw new ConfigurationException("Invalid duration string, found: " + unitString);
-    }
-  }
+  List<String> getCommaSeparatedValues(String name);
 
   /**
-   * Fragments the 'units' portion of a config value from the 'value' portion.
+   * Returns a map-valued configuration property. The format of the original value must be
+   * comma-separated for each key, with an '=' separating the key and value. For instance, <code>
+   * service.name=Greatest Service,host.name=localhost</code> Empty values will be removed.
    *
-   * <p>E.g. "1ms" would return the string "ms".
+   * @return an empty map if the property has not been configured.
+   * @throws ConfigurationException for malformed map strings.
    */
-  private static String getUnitString(String rawValue) {
-    int lastDigitIndex = rawValue.length() - 1;
-    while (lastDigitIndex >= 0) {
-      char c = rawValue.charAt(lastDigitIndex);
-      if (Character.isDigit(c)) {
-        break;
-      }
-      lastDigitIndex -= 1;
-    }
-    // Pull everything after the last digit.
-    return rawValue.substring(lastDigitIndex + 1);
-  }
+  Map<String, String> getCommaSeparatedMap(String name);
```

Comment:
Don't like that `getCommaSeparatedValues` and `getCommaSeparatedMap` return empty list / map instead of null when not set. I know there's code that depends on that behavior, but if we're making a breaking change by changing this class to an interface, now might be a good time to ensure consistency.

Reply:
Hmm, I think we're probably returning nulls for Integers/Longs/etc because there's no "natural null object" for those - with a list or a map you can always return an empty one and avoid unneeded null checks.

Reply:
I don't think we should return nulls. That's a bad API to expose, as all callers will need to do null checking. We should return the empty lists/maps as we do today, IMO.

Reply:
I see where you're coming from, but having responses be nullable allows callers to differentiate between "empty" and "not set". 

Don't feel strongly about this.